src/Agent.Sdk/Knob.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Agent.Sdk
+{
+    public enum KnobSource
+    {
+        BuiltInDefault,
+        EnvironmentVariable,
+        RuntimeVariable,
+    };
+
+    public class KnobValue
+    {
+        public KnobSource Source { get;  private set;}
+        public string Which { get; private set; }
+        private string _value;
+
+        public KnobValue(string value, KnobSource source, string which=null)
+        {
+            _value = value;
+            Source = source;
+            Which = which;
+        }
+
+        public string AsString()
+        {
+            return _value;
+        }
+
+        public bool AsBoolean()
+        {
+            return StringUtil.ConvertToBoolean(_value);
+        }
+    }
+
+    public class AgentKnobs : KnobPanel
+    {
+        public static readonly IKnob UseNode10 = new Knob(nameof(UseNode10))
+        {
+            EnvironmentVariableNames = new List<string>{ "AGENT_USE_NODE10" },
+            RuntimeVariableNames = new List<string>{ "AGENT_USE_NODE10" },
+            Description = "Forces the agent to use Node 10 handler for all Node-based tasks",
+            DefaultValue = "false"
+        };
+    }
+
+    public class KnobPanel
+    {
+
+        private static readonly IDictionary<string,IKnob> _knobs = new Dictionary<string,IKnob>();
+
+        public class Knob : IKnob
+        {
+            public string Name { get; private set; }
+            public List<string> EnvironmentVariableNames { get; set; }
+            public List<string> RuntimeVariableNames { get; set; }
+            // public TBDType CommandLineOption {get; private set;}
+            public string Description { get; set; }
+            public bool IsDeprecated {get; set; } = false;  // is going away at a future date
+            public bool IsExperimental {get; set; } = false; // may go away at a future date
+            public string DefaultValue {get; set; } = "";
+
+            public Knob(string name)
```

Comment:
I'd probably make more fields required in this constructor. Specifically, I don't think we want people registering a knob without a description or defaultValue

Reply:
agreed.